### PR TITLE
Update rollover index API docs for data streams

### PR DESCRIPTION
**Pull request by danhermann**
Documents the behavior of the rollover index API when operating on data streams.

Relates to #53100

### CODE REVIEW

**Review by martijnvg — APPROVED** [review]
LGTM! (pending succesfull build after #55498 gets merged)

**Comment by henningandersen** [comment]
@jrodewig maybe you want to have a look at this too. Given that the terminology (rollover target) will be used in the code too, getting your feedback is valuable.

File: docs/reference/indices/rollover-index.asciidoc
Comment (jrodewig) [inline docs/reference/indices/rollover-index.asciidoc:9]: `indices-datastreams` isn't a valid anchor in the docs right now. This is causing the docs CI to fail.

> or a <<indices-datastreams, data stream>>. When targeting an alias, the alias is updated to

It looks like we don't have data stream docs at all atm. It might be good to add those before merging this.

File: docs/reference/indices/rollover-index.asciidoc
Comment (jrodewig) [inline docs/reference/indices/rollover-index.asciidoc:7]: 👍  I feel like 'rollover target' is a good term for this and it's precisely explained here.

File: docs/reference/indices/rollover-index.asciidoc
Comment (jrodewig) [inline docs/reference/indices/rollover-index.asciidoc:10]: nit: I think "point" is a bit more clear here.
```suggestion
point to the new index. When targeting a data stream, the new index becomes the data stream's
```

File: docs/reference/indices/rollover-index.asciidoc
Comment (jrodewig) [inline docs/reference/indices/rollover-index.asciidoc:41]:
```suggestion
The rollover index API rolls a rollover target's existing index to a new index
when the existing index meets a condition you provide. You can use this API to 
retire an index that becomes too large or too old.
```

File: docs/reference/indices/rollover-index.asciidoc
Comment (jrodewig) [inline docs/reference/indices/rollover-index.asciidoc:85]: For parameters that are required in some situations, we typically add an asterisk.
```suggestion
(Required*, string)
```

File: docs/reference/indices/rollover-index.asciidoc
Comment (jrodewig) [inline docs/reference/indices/rollover-index.asciidoc:135]: Good catch!

File: docs/reference/indices/rollover-index.asciidoc
Comment (jrodewig) [inline docs/reference/indices/rollover-index.asciidoc:220]:
```suggestion
PUT /_data_stream/my-data-stream <1>
```

File: docs/reference/indices/rollover-index.asciidoc
Comment (jrodewig) [inline docs/reference/indices/rollover-index.asciidoc:69]: Should we mention that it's also the write index?

File: docs/reference/indices/rollover-index.asciidoc
Comment (jrodewig) [inline docs/reference/indices/rollover-index.asciidoc:241]: The conditions become a bit hard to parse in a callout. I think breaking them out into a bulleted list might help>
```suggestion
<2> This request creates a new backing index, `my-data-stream-000002`, and adds
it as the write index for the `my-data-stream` data stream if the current
write index meets at least one of the following conditions:
+
--
* The index was created 7 or more days ago.
* The index has an index size of 5GB or greater.
* The index contains 1,000 or more documents.
--
```

File: docs/reference/indices/rollover-index.asciidoc
Comment (jrodewig) [inline docs/reference/indices/rollover-index.asciidoc:305]: nit
```suggestion
If the rollover target is a data stream, you cannot specify a target index
```

**Review by jrodewig — APPROVED** [review]
Look good overall. I mostly left minor wording nits, but feel free to ignore those as you see fit.

I think the rollover target terminology works well, and you did a good job of explaining it concisely upfront.

Here are some other things I wanted to point out:

* We don't currently have any data stream docs. For the best user experience, it may make sense to wait until those are added before merging this PR.
* The `indices-datastreams` anchor linked here doesn't exist. This is likely causing the docs CI build to fail and is related to by first point.
* At several points, the line length exceeds 80 characters. Please rewrap those where possible.

File: docs/reference/indices/rollover-index.asciidoc
Comment (danhermann) [reply docs/reference/indices/rollover-index.asciidoc:9]: @jrodewig, I've added those in https://github.com/elastic/elasticsearch/pull/55557 if you would like to see them. I presume they will need to be merged first to create the appropriate data streams anchors.

**Comment by danhermann** [comment]
@jrodewig, thanks for the detailed review (as well as corrected the optionality of the `conditions` option which was incorrect). I've made your suggested changes. I think I'll need to get https://github.com/elastic/elasticsearch/pull/55551 merged first so the anchors will resolve.

File: docs/reference/indices/rollover-index.asciidoc
Comment (jrodewig) [reply docs/reference/indices/rollover-index.asciidoc]: 👍 Awesome. And yes--- they'll need to be merged in before you can reference the anchor here. You may have to update this branch after #55557 merges to get the docs CI to pass.

**Comment by danhermann** [comment]
@elasticmachine update branch

**Pull request merged**
